cmd/aks-periscope/aks-periscope.go

```diff
@@ -22,25 +23,35 @@ func main() {
 		log.Printf("Failed to create CRD: %+v", err)
 	}
 
-	collectors := []interfaces.Collector{}
+	//create map of all possible collectors by name, discrete vars for each collector as some are diagnoser dependecies
+	allCollectorsByName := make(map[string]interfaces.Collector)
 	containerLogsCollector := collector.NewContainerLogsCollector(exporter)
-	collectors = append(collectors, containerLogsCollector)
+	allCollectorsByName["containerlogs"] = containerLogsCollector
 	systemLogsCollector := collector.NewSystemLogsCollector(exporter)
-	collectors = append(collectors, systemLogsCollector)
+	allCollectorsByName["systemlogs"] = systemLogsCollector
 	networkOutboundCollector := collector.NewNetworkOutboundCollector(5, exporter)
-	collectors = append(collectors, networkOutboundCollector)
+	allCollectorsByName["networkoutbound"] = networkOutboundCollector
 	ipTablesCollector := collector.NewIPTablesCollector(exporter)
-	collectors = append(collectors, ipTablesCollector)
+	allCollectorsByName["iptables"] = ipTablesCollector
 	nodeLogsCollector := collector.NewNodeLogsCollector(exporter)
-	collectors = append(collectors, nodeLogsCollector)
+	allCollectorsByName["nodelogs"] = nodeLogsCollector
 	dnsCollector := collector.NewDNSCollector(exporter)
-	collectors = append(collectors, dnsCollector)
+	allCollectorsByName["dns"] = dnsCollector
 	kubeObjectsCollector := collector.NewKubeObjectsCollector(exporter)
-	collectors = append(collectors, kubeObjectsCollector)
+	allCollectorsByName["kubeobjects"] = kubeObjectsCollector
 	kubeletCmdCollector := collector.NewKubeletCmdCollector(exporter)
-	collectors = append(collectors, kubeletCmdCollector)
+	allCollectorsByName["kubeletcmd"] = kubeletCmdCollector
 	systemPerfCollector := collector.NewSystemPerfCollector(exporter)
-	collectors = append(collectors, systemPerfCollector)
+	allCollectorsByName["systemperf"] = systemPerfCollector
```

Comment:
💡 suggestion: Will this behave the same way? doing this might have all the `New...` object above and the `map string interfaces` below:

```
allCollectorsByName := map[string]interface{}{
  "containerlogs": containerLogsCollector,
  "systemlogs": systemLogsCollector,
  "networkoutbound": networkOutboundCollector,
}
```

Reply:
thanks! Have refactored this